Update list.php

elfibro opened the pull request:
Added condition to include client code in the search when the global variable SOCIETE_ADD_REF_IN_LIST is set to 1.

- Modified the SQL query to include 's.code_client' in the search conditions.
- Added check for the global variable SOCIETE_ADD_REF_IN_LIST.
- If SOCIETE_ADD_REF_IN_LIST is 1, 's.code_client' is included in the search along with 's.nom' and 's.name_alias'.
- Ensured that 's.code_client' is searched separately only when SOCIETE_ADD_REF_IN_LIST is 1 and 'search_societe' is not defined.

# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "FIX", "CLOSE", "NEW", "PERF" or "QUAL" section* (use uppercase to have the PR appears into the ChangeLog, lowercase will not appears)
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed texts with meaningful information*


# FIX|Fix #[*issue_number Short description*]
[*Long description*]


# CLOSE|Close #[*issue_number Short description*]
[*Long description*]


# NEW|New [*Short description*]
[*Long description*]


# PERF|Perf #[*issue_number Short description*]
[*Long description*]


# QUAL|Qual #[*issue_number Short description*]
[*Long description*]

elfibro committed:
Update list.php

Added condition to include client code in the search when the global variable SOCIETE_ADD_REF_IN_LIST is set to 1.

- Modified the SQL query to include 's.code_client' in the search conditions.
- Added check for the global variable SOCIETE_ADD_REF_IN_LIST.
- If SOCIETE_ADD_REF_IN_LIST is 1, 's.code_client' is included in the search along with 's.nom' and 's.name_alias'.
- Ensured that 's.code_client' is searched separately only when SOCIETE_ADD_REF_IN_LIST is 1 and 'search_societe' is not defined.

elfibro committed:
Update list.php

elfibro committed:
Update list.php

eldy commented on htdocs/comm/propal/list.php:
```diff
@@ -635,16 +635,29 @@
 if ($search_availability) {
 	$sql .= " AND p.fk_availability IN (".$db->sanitize($db->escape($search_availability)).')';
 }
+$societe_add_ref_in_list = (!empty($conf->global->SOCIETE_ADD_REF_IN_LIST) ? $conf->global->SOCIETE_ADD_REF_IN_LIST : 0); // Récupérer la valeur de SOCIETE_ADD_REF_IN_LIST
```
Can you replace conf->global->xxx with getDolGlobalString('xxx
Old syntax is no more allowed.

elfibro replied:
It's ok now

elfibro committed:
Update list.php

Refactor: Replace conf->global->SOCIETE_ADD_REF_IN_LIST with getDolGlobalString('SOCIETE_ADD_REF_IN_LIST')

elfibro committed:
Merge branch '20.0' into patch-2

frederic34 commented on htdocs/comm/propal/list.php:
```diff
@@ -635,16 +635,29 @@
 if ($search_availability) {
 	$sql .= " AND p.fk_availability IN (".$db->sanitize($db->escape($search_availability)).')';
 }
+$societe_add_ref_in_list = (!empty(getDolGlobalString('SOCIETE_ADD_REF_IN_LIST')) ? getDolGlobalString('SOCIETE_ADD_REF_IN_LIST') : 0); // Récupérer la valeur de SOCIETE_ADD_REF_IN_LIST
```
`$societe_add_ref_in_list = getDolGlobalInt('SOCIETE_ADD_REF_IN_LIST');` is more simple and pass the tests

elfibro replied:
It's ok now

elfibro committed:
Update list.php

Simplified SOCIETE_ADD_REF_IN_LIST retrieval

Replaced conditional retrieval with getDolGlobalInt('SOCIETE_ADD_REF_IN_LIST') to simplify code and pass tests.

elfibro committed:
Merge branch '20.0' into patch-2
